proposals/NNNN-omit-return.md

```diff
@@ -0,0 +1,361 @@
+# Implicit Returns from Single-Expression Functions
+
+* Proposal: [SE-NNNN](NNNN-omit-return.md)
+* Author: [Nate Chandler](https://github.com/nate_chandler)
+* Review Manager: TBD
+* Status: Implemented
+* Implementation: [apple/swift#23251](https://github.com/apple/swift/pull/23251)
+* Previous Proposal: [SE-NNNN](https://github.com/DevAndArtist/swift-evolution/blob/single_expression_optional_return/proposals/nnnn-single-expression-optional-return.md)
+
+## Introduction
+
+Swift provides a pleasant shorthand for short closures: if a closure contains just a single expression, that expression is implicitly returned--the `return` keyword can be omitted.  We should provide this shorthand for functions as well.
+
+Swift-evolution thread: [Pitch: Implicit Returns from Single-Expression Functions](https://forums.swift.org/t/pitch-implicit-returns-from-single-expression-functions/21898)
+
+## Motivation
+
+Consider the following implementation of the popular `Collection.sum()` extension:
+
+```swift
+extension Sequence where Element == Int {
+
+    func sum() -> Element {
+        return reduce(0, +)
+    }
+
+}
+```
+
+The implementation is extremely brief, weighing in at 19 characters.  Of those 19, however, 7 are consumed by the `return` keyword.  
+
+## Proposed solution
+
+Here's how that same property would look if the single-expression in the body is implicitly returned:
+
+```swift
+func sum() -> Element {
+    reduce(0, +)
+}
+```
+
+The expression which is being returned is front and center, not tucked away behind the `return` keyword.
+
+For readers previously exposed to single-expression closure syntax, this will feel completely familiar.  
+
+Even to readers without such exposure, though, the meaning will be clear: When reading the implementation--after the `var` keyword and name--you first encounter the return type and then the single expression within the body.  Since you've just read the return type, can see that there's only one expression in the body, and are told by the compiler that the code is legal, you are forced to conclude that the expression must indeed be returned.
+
+In fact, exposure to functions--whose types are always stated--without an explicit `return` keyword will help prepare new Swift users to understand code like
+
+```swift
+let ages = persons.map { $0.name }
+```
+
+Their prior exposure to functions which implicitly return the single expression in their bodies will lead them to conclude that the closure being passed to map is returning the expression `$0.name` and that the return type of the closure is `String` (the type of `name`, here).
+
+## Detailed design
+
+Interpret the bodies of function-like entities which consist of a single expression as returning that expression--unless the entity's return type is `Void` or the single expression's type is uninhabited.
+
+#### Function-Like Entities
```

Comment:
You bounce back and forth between title case and sentence case for your headings. This heading and the proposal title are title case, the rest are sentence case.

Reply:
Nice catch.  As of https://github.com/apple/swift-evolution/pull/1015/commits/32f9fb3fdf0a8f0c3b5aea8ba2aaa54712a36421 they're all sentence case.